src/foam/core/Property.js

```diff
@@ -308,6 +308,10 @@ foam.CLASS({
       if ( this.forClass_ && this.forClass_ !== c.id && prop === this ) {
         // Clone this property if it's been installed before.
         prop = this.clone();
+
+        // sourceCls_ isn't a real property so it gets lost during the clone.
+        prop.sourceCls_ = c; 
```

Comment:
This seems right but nothing should be broken in its absence because we shouldn't really be relying on this property. The comment around it here: https://github.com/foam-framework/foam2/blob/11463bc73914489079a9781ca148bf82b8531b75/src/foam/core/FObject.js#L149 says it's for the describe method.

Reply:
I don't think this is necessary.  installAxiom() sets it, and you're calling installAxiom in your AxiomClone.installInClass().

This code path shouldn't even be hit.  forClass_ shouldn't be set on the cloned axiom that AxiomCloner creates.  We maybe need to clear it, or update the logic on clone() or have a specific transient type for properties such that they don't get cloned.  Maybe override cloneProperty for forClass_ would be easiest.

Reply:
forClass_ does get copied over since it's a real property on foam.core.Property. I don't think cloning looks at transient properties. And sourceCls_ gets set during installAxiom but gets cleared when installAxiom calls installInClass.

Reply:
I'm saying we _shouldn't_ copy over forClass_, not that we don't copy over forClass_.

Reply:
Chatted offline about this to clarify. This is just a fix for an unrelated bug.